### PR TITLE
Full screen opaque toolbar issue (closes #249)

### LINKED ISSUE #249: Incorrect rendering of titlebar in full screen mode
**Description**

The demo app, when switched to Full Screen Mode, displays an empty titlebar. See the screenshot attached.

**Steps To Reproduce**

1. Compile the demo.
2. Launch the demo.
3. Enter the full screen mode.

**Expected behavior**

The upper part of the sidebar and the upper part of the main pane should still be visible.

**Screenshots**

Capture d’écran 2022-05-13 à 12 45 05

### PR DESCRIPTION
**Pull request by whiplashoo**
Fixes #249 by editing the app window's implementation in Swift. Now when the app goes in full screen, no toolbar is shown at all. When it returns from full screen, the toolbar is again "visible", to make the close/minimize/maximize buttons have the proper alignment.

https://user-images.githubusercontent.com/9117427/168389835-70cd2318-26f3-4172-835d-fba234857333.mp4

The docs have also been updated.

## Pre-launch Checklist

- [👍 ] I have run `dartfmt` on all changed files 
- [ 👍 ] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [👍  ] I have added/updated relevant documentation 
- [👍  ] I have run "optimize/organize imports" on all changed files
- [👍  ] I have addressed all analyzer warnings as best I could

### CODE REVIEW

**Comment by GroovinChip** [comment]
I'm a little confused about the language you're using to describe what's happening here. As far as I can tell, the Toolbar widget remains visible in fullscreen. Are you referring to native titlebar?

**Comment by whiplashoo** [comment]
Yes, I am referring to the native element which we build [here](https://github.com/GroovinChip/macos_ui/blob/ab47743b6d86134afb0134281b390a5d8517469a/example/macos/Runner/MainFlutterWindow.swift#L45) (to have the proper alignment for the minimize/maximize/close buttons). Our `ToolBar` widget remains now visible as intended, yes.

**Comment by GroovinChip** [comment]
Okay, understood. Thanks for the clarification.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## [1.4.1]
+* Fixes an issue where if the app was displayed in full screen mode, an opaque empty toolbar would appear at the top [#249](https://github.com/GroovinChip/macos_ui/issues/249)
```
Comment (GroovinChip) [inline CHANGELOG.md:2]: I know in the code we're creating an `NSToolbar`, but shouldn't it be called a titlebar? From what I know of cocoa/appkit applications, that part of the window is referred to as a titlebar. I could be wrong, though.

**Review by GroovinChip — DISMISSED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (whiplashoo) [reply CHANGELOG.md:2]: Screenshot 2022-05-14 at 11 50 45 PM

As I understand, the title bar is the thin bar that holds the apple icon and the menu. In our issue, the title bar would still appear when you hovered the mouse at the top of the full screen window (and still does). However, the empty toolbar from `NSToolbar` appeared right below it, obscuring the top part of the app's window.

Comment (GroovinChip) [reply CHANGELOG.md:2]: Alright, then.

**Review by GroovinChip — APPROVED** [review]
LGTM

**Comment by GroovinChip** [comment]
Actually hang on, why are the checks failing? I'm on mobile.

**Pull request merged**